sdl-core/src/test/scala/io/smartdatalake/workflow/action/CustomDataFrameActionTest.scala

```diff
@@ -399,6 +401,95 @@ class CustomDataFrameActionTest extends FunSuite with BeforeAndAfter {
     // getSparkDataFrame will throw IllegalArgumentException because it no files have been written to this DataObject...
     intercept[IllegalArgumentException](tgtDO2.getSparkDataFrame())
   }
+
+  test("copy load with constraints and expectations non-main input no_data") {
+
+    // setup DataObjects
+    val srcDO1 = MockDataObject("src1", expectations = Seq(
+      CountExpectation(name = "count", expectation = Some(">= 1"))
+    )).register
+    val srcDO2 = MockDataObject("src2", expectations = Seq(
+      CountExpectation(name = "count", expectation = Some("= 0")),
+      CountExpectation(name = "countAll", expectation = Some("= 0"), scope = ExpectationScope.All)
+    )).register
+    val tgtDO1 = MockDataObject("tgt1", expectations = Seq(
+      CountExpectation(expectation = Some(">= 1")),
+      SQLExpectation("tgt1AvgRatingGt1", Some("avg rating should be bigger than 1"), "avg(rating)", Some("> 1")),
+    )).register
+    val tgtDO2 = MockDataObject("tgt2", expectations = Seq(
+      CountExpectation(expectation = Some("= 0")),
+      SQLExpectation("tgt2AvgRatingGt1", Some("avg rating should be bigger than 1"), "avg(rating)", Some("> 1")),
+    )).register
+
+    // prepare & start load
+    val customTransformerConfig1 = SQLDfsTransformer(code = Map(tgtDO1.id.id -> "select * from src1"))
+    val customTransformerConfig2 = SQLDfsTransformer(code = Map(tgtDO2.id.id -> "select * from src2"))
+    val action1 = CustomDataFrameAction("ca", List(srcDO1.id, srcDO2.id), List(tgtDO1.id, tgtDO2.id), mainInputId = Some(srcDO1.id), mainOutputId = Some(tgtDO1.id),
+      transformers = Seq(customTransformerConfig1, customTransformerConfig2),
+      expectations = Seq(TransferRateExpectation(), CompletnessExpectation(expectation = None))
+    )
+    instanceRegistry.register(action1)
+    val l = Seq(("jonson","rob",5),("doe","bob",3)).toDF("lastname", "firstname", "rating")
```

Comment:
can you rename this? Maybe expectedResult?

Reply:
Ok, done